routers/api/v1/packages/package.go

```diff
@@ -211,3 +211,52 @@ func ListPackageFiles(ctx *context.APIContext) {
 
 	ctx.JSON(http.StatusOK, apiPackageFiles)
 }
+
+// LinkPackage sets a repository link for a package
+func LinkPackage(ctx *context.APIContext) {
+	// swagger:operation POST /packages/{owner}/{type}/{name}/link package linkPackage
+	// ---
+	// summary: Link a package to a repository
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the package
+	//   type: string
+	//   required: true
+	// - name: type
+	//   in: path
+	//   description: type of the package
+	//   type: string
+	//   required: true
+	// - name: name
+	//   in: path
+	//   description: name of the package
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: query
+	//   description: ID of the repository to link
+	//   type: integer
+	//   required: true
+	// responses:
+	//   "201":
+	//     "$ref": "#/responses/empty"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	repoID := ctx.FormInt64("repo")
+	packageType := ctx.Params("type")
+	name := ctx.Params("name")
+	pkg, err := packages.GetPackageByName(ctx, ctx.ContextUser.ID, packages.Type(packageType), name)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "LinkPackage", err)
+		return
+	}
+
+	err = packages_service.LinkPackageToRepository(ctx, ctx.Doer, pkg, repoID)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "LinkPackage", err)
+		return
+	}
+	ctx.Status(http.StatusNoContent)
```

Comment:
```suggestion
	pkg, err := packages.GetPackageByName(ctx, ctx.ContextUser.ID, packages.Type(ctx.Params("type")), ctx.Params("name"))
	if err != nil {
		if errors.Is(err, utils.ErrNotExist) {
			ctx.Error(http.StatusNotFound, "GetPackageByName", err)
		} else {			
			ctx.Error(http.StatusInternalServerError, "GetPackageByName", err)
		}
		return
	}

	err = packages_service.LinkPackageToRepository(ctx, ctx.Doer, pkg, ctx.FormInt64("repo"))
	if err != nil {
		ctx.Error(http.StatusInternalServerError, "LinkPackage", err)
		return
	}
	ctx.Status(http.StatusNoContent)
```
And it needs an additional import for `errors` then.